openwisp_monitoring/monitoring/base/models.py

```diff
@@ -550,32 +607,25 @@ def _time_crossed(self, time):
         threshold_time = timezone.now() - timedelta(minutes=self.tolerance)
         return time < threshold_time
 
-    def _is_crossed_by(self, current_value, time=None, retention_policy=None):
-        """ do current_value and time cross the threshold? """
-        value_crossed = self._value_crossed(current_value)
-        if value_crossed is NotImplemented:
-            raise ValueError('Supplied value type not suppported')
-        if not value_crossed:
```

Comment:
maybe it's enough to just add ` and self.tolerance == 0` and make sure the rest of the code of this function works also when the value does not cross the threshold, that is:

- if tolerance is not zero
- return False only if the all the recent values do not cross the threshold

Right now instead, the code returns False as soon as hits `not value_crossed`

Reply:
I tried that approach in the beginning and it didn't seem to work but I will consider reviewing it again